test/browser-tests/router-get/router-get.js

```diff
@@ -0,0 +1,296 @@
+/*
+  Copyright 2016 Google Inc. All Rights Reserved.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+// This is a test and we want descriptions to be useful, if this
+// breaks the max-length, it's ok.
+
+/* eslint-disable max-len */
+/* eslint-env browser, mocha */
+/* global testHelper */
+
+'use strict';
+
+describe('Test router.get method', () => {
+  let performFetch = (iframe, fetchUrl, expectedString) => {
+    return iframe.contentWindow.fetch(fetchUrl)
+      .then(response => {
+        response.status.should.equal(200);
+        return response.text();
+      })
+      .then(responseText => {
+        responseText.should.equal(expectedString);
+      });
+  };
+
+  let performTest = (swUrl, fetchUrl, expectedString, done) => {
+    let iframe = null;
+    let testPromise = testHelper.activateSW(swUrl)
+    .then(newIframe => {
+      iframe = newIframe;
+
+      // Call the iframes fetch event so it goes through the service worker
+      return performFetch(newIframe, fetchUrl, expectedString);
+    });
+
+    if (done) {
+      testPromise = testPromise.then(() => done(), done);
+    }
+
+    testPromise = testPromise.then(() => iframe);
```

Comment:
I don't understand what this accomplishes. Could you add a comment clarifying why this is needed?